### PR TITLE
implement Logger

### PR DESCRIPTION
**Pull request by fgrunewald**
Here is the long sought after logger.

### CODE REVIEW

File: bin/polyply [hunk 1]
```diff
@@ -58,6 +84,9 @@ def main():
 
     parser_gen_itp.add_argument('-name', required=True, type=str, dest="name",
                                 help="name of the final molecule")
+    parser_gen_itp.add_argument('-v', dest='verbosity', action='count',
+                             help='Enable debug logging output. Can be given '
+                                  'multiple times.', default=0)
```
Comment (pckroon) [inline bin/polyply:89]: lint

File: bin/polyply [hunk 2]
```diff
@@ -81,6 +110,10 @@ def main():
     parser_gen_coords.add_argument('-name', required=True, type=str, dest="name",
                                    help="name of the final molecule")
 
+    parser_gen_coords.add_argument('-v', dest='verbosity', action='count',
+                             help='Enable debug logging output. Can be given '
+                                  'multiple times.', default=0)
+
```
Comment (pckroon) [inline bin/polyply:116]: lint. And can't these be deduplicated?

File: bin/polyply [hunk 3]
```diff
@@ -159,6 +192,9 @@ def main():
 
     parser_gen_seq.add_argument('-name', required=True, type=str, dest="name",
                                 help="name of the final molecule")
+    parser_gen_seq.add_argument('-v', dest='verbosity', action='count',
+                             help='Enable debug logging output. Can be given '
+                                  'multiple times.', default=0)
```
Comment (pckroon) [inline bin/polyply:197]: .

File: polyply/src/gen_coords.py [hunk 4]
```diff
@@ -81,26 +84,28 @@ def _check_molecules(molecules):
 
 def gen_coords(args):
     # Read in the topology
-    # TODO Write Logger for all print statements
-    print("INFO - reading topology")
+    # TODO Write LOGGER for all print statements
```
Comment (pckroon) [inline polyply/src/gen_coords.py:87]: This one can now be resolved, no?

File: polyply/src/generate_templates.py [hunk 5]
```diff
@@ -308,9 +311,9 @@ def _gen_templates(self, meta_molecule):
                     if success:
                         break
                     elif opt_counter > self.max_opt:
-                        print("WARNING: Failed to optimize structure for block {}.".format(resname))
-                        print("         Proceeding with unoptimized coordinates.")
-                        print("         Usually this is OK, but check your final structure.")
+                        LOGGER.warning(("WARNING: Failed to optimize structure for block {}."
+                                        "Proceeding with unoptimized coordinates."
+                                        "Usually this is OK, but check your final structure.", resname))
```
Comment (pckroon) [inline polyply/src/generate_templates.py:316]: Remove the WARNING from the string. That should now be taken care of by the logging formatter

**Review by pckroon — CHANGES_REQUESTED** [review]
Few small nitpicks, but looks good otherwise

File: bin/polyply (hunk 2 above)
Comment (fgrunewald) [reply bin/polyply:116]: Not worth it considering the complexity of the parser already.

File: polyply/src/gen_coords.py (hunk 4 above)
Comment (fgrunewald) [reply polyply/src/gen_coords.py:87]: Indeed it is resolved.

**Review by pckroon — APPROVED** [review]

**Pull request merged**